supportApp/videoCompressionSrc/video_compression.h

```diff
@@ -0,0 +1,20 @@
+#ifndef VIDEO_COMPRESSION_H
+#define VIDEO_COMPRESSION_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+int H264_compress(const char* source, char* dest, int x_size, int y_size);
```

Comment:
Shouldn't the size for `dest` be passed to this function?

I'd also feel more comfortable with the type in use being `unsigned char` rather than `char`. Pixels won't have negative values after all.

Reply:
dest is the buffer where the compressed image is stored, but the buffer itself is given the same size as the uncompressed image. H264_compress returns the number of bytes written to dest. If the compressed image is for some reason larger than the uncompressed image, the current behavior is to consider it a failure and reject the compressed image.

Reply:
The only reason H264_compress would need the size of dest would be if it did the checking to see if the dest buffer is large enough. The H264_compress function is currently implemented under the assumption that the dest buffer is large enough to store the compressed image. Whether it is large enough depends on the upper bound function being accurate enough (which I admit this is unknown to me), since this is what ensures that dest is large enough. So far I have not noticed issues here, but I will do testing to further investigate this.